typescript/sdk/src/ism/EvmIsmModule.ts

```diff
@@ -86,4 +186,610 @@ export class EvmIsmModule extends HyperlaneModule<
+          destination,
+          config,
+          logger,
+          multiProvider,
+          factories,
+        });
+        break;
+      case IsmType.ROUTING:
+      case IsmType.FALLBACK_ROUTING:
+        contract = await EvmIsmModule.deployRoutingIsm({
+          destination,
+          config,
+          origin,
+          mailbox,
+          logger,
+          multiProvider,
+          factories,
+          deployer,
+        });
+        break;
+      case IsmType.AGGREGATION:
+        contract = await EvmIsmModule.deployAggregationIsm({
+          destination,
+          config,
+          origin,
+          mailbox,
+          logger,
+          multiProvider,
+          factories,
+          deployer,
+        });
+        break;
+      case IsmType.OP_STACK:
+        contract = await deployer.deployContractFromFactory(
+          destination,
+          new OPStackIsm__factory(),
+          IsmType.OP_STACK,
+          [config.nativeBridge],
+        );
+        break;
+      case IsmType.PAUSABLE:
+        contract = await deployer.deployContractFromFactory(
+          destination,
+          new PausableIsm__factory(),
+          IsmType.PAUSABLE,
+          [
+            await resolveOrDeployAccountOwner(
+              multiProvider,
+              destination,
+              config.owner,
+            ),
+          ],
+        );
+        await deployer.transferOwnershipOfContracts(destination, config, {
+          [IsmType.PAUSABLE]: contract,
+        });
+        break;
+      case IsmType.TRUSTED_RELAYER:
+        assert(mailbox, `Mailbox address is required for deploying ${ismType}`);
+        contract = await deployer.deployContractFromFactory(
+          destination,
+          new TrustedRelayerIsm__factory(),
+          IsmType.TRUSTED_RELAYER,
+          [mailbox, config.relayer],
+        );
+        break;
+      case IsmType.TEST_ISM:
+        contract = await deployer.deployContractFromFactory(
+          destination,
+          new TestIsm__factory(),
+          IsmType.TEST_ISM,
+          [],
+        );
+        break;
+      default:
+        throw new Error(`Unsupported ISM type ${ismType}`);
+    }
+
+    return contract;
+  }
+
+  protected static async deployMultisigIsm(params: {
+    destination: ChainName;
+    config: MultisigIsmConfig;
+    logger: Logger;
+    multiProvider: MultiProvider;
+    factories: HyperlaneAddresses<ProxyFactoryFactories>;
+  }): Promise<IMultisigIsm> {
+    const { destination, config, logger, multiProvider, factories } = params;
+
+    const signer = multiProvider.getSigner(destination);
+    const multisigIsmFactory =
+      config.type === IsmType.MERKLE_ROOT_MULTISIG
+        ? factories.staticMerkleRootMultisigIsmFactory
+        : factories.staticMessageIdMultisigIsmFactory;
+    const factory: StaticThresholdAddressSetFactory =
+      StaticThresholdAddressSetFactory__factory.connect(
+        multisigIsmFactory,
+        signer,
+      );
```

Comment:
can we leverage the connected factories if this is not static?

Reply:
yup

but then it's a question of how we want the module interaction to flow if Module.create() is no longer static

Reply:
yes - but then the flow for module users would be 

```
const module = Module.create()
module.deploy()
module.create()
module.update()
...
```

maybe the actual interface we desire is not create/read/update but deploy/read/update, and we do Module.new() to instantiate a class for an undeployed module?